packages/sdk/src/interaction/interaction.js

```diff
@@ -149,9 +159,16 @@ export const prepAccount = (acct, opts = {}) => ix => {
   const role = opts.role
   const tempId = uuid()
 
-  if (acct.authorization && isFn(acct.authorization)) acct = {resolve: acct.authorization}
+  if (acct.authorization && isFn(acct.authorization))
+    acct = {resolve: acct.authorization}
   if (!acct.authorization && isFn(acct)) acct = {resolve: acct}
 
+  const resolve = acct.resolve
+  if (resolve)
+    acct.resolve = acct =>
+      [resolve, prepAccountKeyId].reduce(async (d, fn) => fn(await d), acct)
```

Comment:
`resolve` is a function that consumes multiple arguments.

See: https://github.com/onflow/fcl-js/blob/master/packages/sdk/src/resolve/resolve-accounts.js#L30


We'll need to modify this to support multiple args. Perhaps something like:

```suggestion
     acct.resolve = (acct, ...rest) =>  [resolve, prepAccountKeyId].reduce(async (d, fn) => fn(await d, ...rest), acct)
```

^ Please check that that works correctly though, i'm not 100% confident with it

Reply:
It looks good to me.  Just tested it to be sure and it is all good.